src/backend/distributed/executor/local_executor.c

```diff
@@ -760,6 +760,15 @@ ShouldExecuteTasksLocally(List *taskList)
 		 */
 		return true;
 	}
+	else if (ShouldRunTasksSequentially(taskList) && AnyTaskAccessesLocalNode(taskList))
```

Comment:
We had a better solution in https://github.com/citusdata/citus/pull/4374, so un-doing this change.